protocol/messenger_unread_test.go

```diff
@@ -0,0 +1,277 @@
+	s.Require().Equal(false, actualChat.Highlight)
+
+	actualMessage1, err = s.m.MessageByID(inputMessage1.ID)
+	s.Require().NoError(err)
+	s.Require().Equal(true, actualMessage1.Seen)
+
+	actualMessage2, err = s.m.MessageByID(inputMessage2.ID)
+	s.Require().NoError(err)
+	s.Require().Equal(false, actualMessage2.Seen)
+
+	actualMessage3, err = s.m.MessageByID(inputMessage3.ID)
+	s.Require().NoError(err)
+	s.Require().Equal(false, actualMessage3.Seen)
+}
+
+func (s *MessengerSuite) TestMarkMessageAsUnreadWhenMessageIsAlreadyInUnreadState() {
+	chat := CreatePublicChat("test-chat-3", s.m.transport)
+
+	chat.UnviewedMessagesCount = 1
+	chat.UnviewedMentionsCount = 0
+	chat.Highlight = true
+
+	err := s.m.SaveChat(chat)
+	s.Require().NoError(err)
+
+	inputMessage1 := buildTestMessage(*chat)
+	inputMessage1.ID = "1"
+	inputMessage1.Seen = false
+	inputMessage1.Mentioned = false
+
+	err = s.m.SaveMessages([]*common.Message{inputMessage1})
+	s.Require().NoError(err)
+
+	chats := s.m.allChats
+	actualChat, ok := chats.Load(chat.ID)
+
+	s.Require().Equal(true, ok)
+	s.Require().Equal(uint(1), actualChat.UnviewedMessagesCount)
+	s.Require().Equal(uint(0), actualChat.UnviewedMentionsCount)
+	s.Require().Equal(true, actualChat.Highlight)
+
+	actualMessage, err := s.m.MessageByID(inputMessage1.ID)
+	s.Require().NoError(err)
+	s.Require().Equal(false, actualMessage.Seen)
+
+	count, countWithMentions, err := s.m.MarkMessageAsUnread(chat.ID, inputMessage1.ID)
+	s.Require().NoError(err)
+	s.Require().Equal(uint64(1), count)
+	s.Require().Equal(uint64(0), countWithMentions)
+
+	chats = s.m.allChats
+	actualChat, ok = chats.Load(chat.ID)
+
+	s.Require().Equal(true, ok)
+	s.Require().Equal(uint(1), actualChat.UnviewedMessagesCount)
+	s.Require().Equal(uint(0), actualChat.UnviewedMentionsCount)
+	s.Require().Equal(false, actualChat.Highlight)
+
+	actualMessage, err = s.m.MessageByID(inputMessage1.ID)
+	s.Require().NoError(err)
+	s.Require().Equal(false, actualMessage.Seen)
+}
+
+func (s *MessengerSuite) TestMarkMessageAsUnreadInOneChatDoesNotImpactOtherChats() {
+	chat1 := CreatePublicChat("test-chat-1", s.m.transport)
+	err := s.m.SaveChat(chat1)
+	s.Require().NoError(err)
+
+	chat2 := CreatePublicChat("test-chat-2", s.m.transport)
+	err = s.m.SaveChat(chat2)
+	s.Require().NoError(err)
+
+	chat1.UnviewedMessagesCount = 2
+	chat1.UnviewedMentionsCount = 0
+	chat1.Highlight = true
+
+	inputMessage1 := buildTestMessage(*chat1)
+	inputMessage1.ID = "1"
+	inputMessage1.Seen = false
+	inputMessage1.Mentioned = false
+
+	inputMessage2 := buildTestMessage(*chat1)
+	inputMessage2.ID = "2"
+	inputMessage2.Seen = false
+	inputMessage2.Mentioned = false
+
+	err = s.m.SaveMessages([]*common.Message{inputMessage1, inputMessage2})
+	s.Require().NoError(err)
+
+	chat2.UnviewedMessagesCount = 1
+	chat2.UnviewedMentionsCount = 0
+	chat2.Highlight = true
+
+	inputMessage3 := buildTestMessage(*chat2)
+	inputMessage3.ID = "3"
+	inputMessage3.Seen = false
```

Comment:
hmmm, but it's `false` from the start of the test 🤔  Shouldn't it be `true` here?
And same for `inputMessage1` and `inputMessage2`.

Reply:
This is chat2, so it's never touched 😄 so, it should be false, not true